attestation-service/verifier/src/az_snp_vtpm/mod.rs

```diff
@@ -106,49 +123,96 @@ fn verify_snp_report(
+    const SIGNATURE: &[u8; 256] = include_bytes!("../../test_data/az-vtpm-quote-sig.bin");
+    const MESSAGE: &[u8; 122] = include_bytes!("../../test_data/az-vtpm-quote-msg.bin");
+
     #[test]
     fn test_verify_snp_report() {
-        let report = include_bytes!("../../test_data/az-hcl-data.bin");
-        let hcl_data: HclData = report.as_slice().try_into().unwrap();
+        let hcl_report = HclReport::new(REPORT.to_vec()).unwrap();
+        let snp_report = hcl_report.try_into().unwrap();
         let vcek = Vcek::from_pem(include_str!("../../test_data/az-vcek.pem")).unwrap();
         let vendor_certs = load_milan_cert_chain().as_ref().unwrap();
-        verify_snp_report(hcl_data.report().snp_report(), &vcek, vendor_certs).unwrap();
-
-        let mut wrong_report = *report;
+        verify_snp_report(&snp_report, &vcek, vendor_certs).unwrap();
+    }
 
+    #[test]
+    fn test_verify_snp_report_failure() {
+        let mut wrong_report = REPORT.clone();
         // messing with snp report
         wrong_report[0x00b0] = 0;
-        let wrong_hcl_data: HclData = wrong_report.as_slice().try_into().unwrap();
+        let hcl_report = HclReport::new(wrong_report.to_vec()).unwrap();
+        let snp_report = hcl_report.try_into().unwrap();
+        let vcek = Vcek::from_pem(include_str!("../../test_data/az-vcek.pem")).unwrap();
         let vendor_certs = load_milan_cert_chain().as_ref().unwrap();
-        verify_snp_report(wrong_hcl_data.report().snp_report(), &vcek, vendor_certs).unwrap_err();
+        verify_snp_report(&snp_report, &vcek, vendor_certs).unwrap_err();
+    }
+
+    #[test]
+    fn test_verify_report_data() {
+        let hcl_report = HclReport::new(REPORT.to_vec()).unwrap();
+        let var_data_hash = hcl_report.var_data_sha256();
+        let snp_report = hcl_report.try_into().unwrap();
+        verify_report_data(&var_data_hash, &snp_report).unwrap();
+    }
+
+    #[test]
+    fn test_verify_report_data_failure() {
+        let mut wrong_report = REPORT.clone();
+        wrong_report[0x06e0] += 1;
+        let hcl_report = HclReport::new(wrong_report.to_vec()).unwrap();
+        let var_data_hash = hcl_report.var_data_sha256();
+        let snp_report = hcl_report.try_into().unwrap();
+        verify_report_data(&var_data_hash, &snp_report).unwrap_err();
     }
 
     #[test]
     fn test_verify_quote() {
-        let signature = include_bytes!("../../test_data/az-vtpm-quote-sig.bin").to_vec();
-        let message = include_bytes!("../../test_data/az-vtpm-quote-msg.bin").to_vec();
-        let quote = Quote { signature, message };
-        let report = include_bytes!("../../test_data/az-hcl-data.bin");
-        let hcl_data: HclData = report.as_slice().try_into().unwrap();
+        let quote = Quote {
+            signature: SIGNATURE.to_vec(),
+            message: MESSAGE.to_vec(),
+        };
+        let hcl_report = HclReport::new(REPORT.to_vec()).unwrap();
         let nonce = "challenge".as_bytes();
```

Comment:
So this value is baked into the `message` that is loaded from binary above? Maybe this should also be a constant? Might be clearer to call it `report_data` rather than nonce. Not really a big deal either way.